src/ElementComparator.h

```diff
@@ -0,0 +1,84 @@
+//
+//  ElementComparator.h
+//  drafter
+//
+//  Created by Jiri Kratochvil on 28/02/18.
+//  Copyright (c) 2018 Apiary Inc. All rights reserved.
+//
+#ifndef DRAFTER_ELEMENTCOMPARATOR_H
+#define DRAFTER_ELEMENTCOMPARATOR_H
+
+#include "refract/InfoElements.h"
+#include <set>
+#include <algorithm>
+
+namespace drafter
+{
+    using namespace refract;
+    struct InfoElementsComparator {
+        bool operator()(const InfoElements& rhs, const InfoElements& lhs)
+        {
+
+            const std::set<std::string> ignore = { "sourceMap" };
+            std::set<std::string> keys;
+
+            if (std::find_if(lhs.begin(),
+                    lhs.end(),
+                    [&keys, &rhs, &ignore](const auto& info) {
+                        auto key = info.first;
+                        if (ignore.find(key) != ignore.end()) {
+                            return false;
+                        }
+                        keys.insert(key);
```

Comment:
We can not compare ranges. For example, first element can contain  `sourceMap` while second one not. For this case we have no same count of attrs/meta while elements can still be recognized as equal.

For this reason we compare key by key. And while comparation happen we save set of already compred `keys` into set, to avoid compare them again on second element.

So in general I do not modify stack, just save some temporary info.

Reply:
`keys` is living on the stack and you are calling `insert` on it, which is non-`const`. That's why the lambda modifies the stack. But that's actually not the point.

Quoting cppreference, `find_if` suggests you are searching for something:
```
find finds the first element satisfying specific criteria
```
Something described by the **predicate** you fed it, specifically:
```
find_if searches for an element for which predicate p returns true
```
That's how every other C++ programmer will try to read this code. But the lambda you feed `find_if` is nothing like a predicate, and is modifying values captured from the stack. Yes it works, I'm just saying, it's not easy to read because it abuses find_if.

Reply:
What about a custom iterator skipping `sourceMap` entries to make it more readable?

Reply:
If you look inside, it is predicate. 
1. look if it is in ignored 
2. look if there is same key in rhs
3. check if given(lhs) and found(rhs) values are same.

I save already comparated keys to avoid check them in rhs.

Similar algo is used to check all rhs.

But ok. I rewrite it